ui_framework/components/modal/modal.js

```diff
@@ -0,0 +1,14 @@
+import React from 'react';
+
+export function KuiModal({ children, ...rest }) {
+  return (
+    <div className="kuiModal" { ...rest }>
```

Comment:
FYI, these required html props, as far as I can tell, have to be handled somewhat specially because of the `-` in their name.  So instead of doing:
```
  KuiConfirmModal({ stuff, aria-label, data-test-subj}) {
 ....
```
You have to do:
```
  KuiConfirmModal({ stuff, ...rest }) {
  const ariaLabel = rest['aria-label'];
  const dataTestSubj = rest['data-test-subj'];
 ....
```

Unless anyone else knows a better way to handle that?